metricflow/query/query_parser.py

```diff
@@ -172,6 +176,65 @@ def _top_fuzzy_matches(
         top_ranked_suggestions = [x for x in top_ranked_suggestions if x[1] > min_score]
         return [x[0] for x in top_ranked_suggestions]
 
+    def parse_and_validate_saved_query(
+        self,
+        saved_query_parameter: SavedQueryParameter,
+        where_filter: Optional[WhereFilter],
+        limit: Optional[int],
+        order_by_parameters: Optional[Sequence[OrderByQueryParameter]],
+    ) -> MetricFlowQuerySpec:
+        """Parse and validate a query using parameters from a pre-defined / saved query.
+
+        Additional parameters act in conjunction with the parameters in the saved query.
+        """
+        saved_query = self._get_saved_query(saved_query_parameter)
+
+        # This logic could be encapsulated in the WhereFilter through a merge interface.
+        where_conditions: List[str] = []
+        if saved_query.where is not None:
+            where_conditions.extend(
+                tuple(saevd_query_where_filter.where_sql_template for saevd_query_where_filter in saved_query.where)
+            )
+        if where_filter is not None:
+            where_conditions.append(where_filter.where_sql_template)
+
+        if len(where_conditions) == 0:
+            combined_where_filter = None
+        if len(where_conditions) == 1:
+            combined_where_filter = PydanticWhereFilter(where_sql_template=where_conditions[0])
+        else:
+            where_conditions_with_parenthesis = tuple(f"({where_condition})" for where_condition in where_conditions)
+            combined_where_filter = PydanticWhereFilter(
+                where_sql_template=" AND ".join(where_conditions_with_parenthesis)
+            )
```

Comment:
eeeewwwwww......

I was thinking about eventually moving this to the renderers, since we won't always render the combined filter. The container class I created should help with this.

Reply:
The container class helped, but I didn't see a place to render it?

Reply:
It's currently here: https://github.com/dbt-labs/metricflow/pull/809/files#diff-14f344e3a9bd66bc0d46537a8017023c16cd242d65ace611485b8277e0719b74

Reply:
Yeah I was thinking eventually we'd just use the container and pass it all the way through but maybe that's not the right way to do it. What you did in #809 seems reasonable for now.